### PR TITLE
extend cymbol command to support struct loading from header files

### PR DESCRIPTION
**Pull request by audi99**
This update should allow for (hopefully) easier debugging sessions by enhancing the cymbol command, incorporating external struct information directly into pwndbg, facilitating easier analysis of data structures without the need for source code modification.

- Parsing of C header files to extract struct definitions.
- Configuration options to specify custom header file paths.

### CODE REVIEW

File: pwndbg/commands/cymbol.py [hunk 1]
```diff
@@ -141,6 +141,24 @@ def add_custom_structure(custom_structure_name: str) -> None:
...
+
+    if not custom_structures_source:
+        print(message.notice("The header file is empty or does not contain valid structures."))
+        return
+
+    custom_structure_name = os.path.splitext(os.path.basename(header_file_path))[0]
+    pwndbg_custom_structure_path = os.path.join(pwndbg_cachedir, custom_structure_name) + ".c"
+
+    with open(pwndbg_custom_structure_path, "w") as f:
+        f.write(custom_structures_source)
```
Comment (disconnect3d) [inline pwndbg/commands/cymbol.py:156]: What is this exactly for / why do we need to write the file back again?

File: pwndbg/commands/cymbol.py [hunk 2]
```diff
@@ -247,6 +265,14 @@ def show_custom_structure(custom_structure_name: str, custom_structure_path: str
     default=None,
     type=str,
 )
+parser.add_argument(
+    "-f",
+    "--file",
```
Comment (disconnect3d) [inline pwndbg/commands/cymbol.py:270]: This argument is not used at all.

**Review by disconnect3d — CHANGES_REQUESTED** [review]
This PR seems incomplete, did u forget to commit some code?

**Comment by disconnect3d** [comment]
Ping @audi99

File: pwndbg/commands/cymbol.py (hunk 1 above)
Comment (audi99) [reply pwndbg/commands/cymbol.py:156]: it was to allow for user's custom structures to persist for reuse across sessions, but if this is unneeded and/or redundant i can toss it

**Comment by audi99** [comment]
> Ping @audi99

apologies for the late response, i work nights so my schedule is wack. should be fixed

**Pull request closed without merging**